zephyr/samples/modules/mcuboot/mcuboot_external_config/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2022 Legrand North America, LLC.
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+
+
+set(APPLICATION_CONFIG_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../../../../boot/zephyr)
+
+set(CONF_FILE
+  "${CONF_FILE};${APPLICATION_CONFIG_DIR}/prj.conf;prj.conf"
+  CACHE STRING "" FORCE
+  )
+
+
+# Add a common dts overlay necessary to ensure mcuboot is linked into,
+# and fits inside, the boot partition. (If the user specified a
+# DTC_OVERLAY_FILE on the CMake command line, we need to append onto
+# the list).
+set(DTC_OVERLAY_FILE
+  "${DTC_OVERLAY_FILE};${APPLICATION_CONFIG_DIR}/dts.overlay"
+  CACHE STRING "" FORCE
+  )
```

Comment:
setting CONF_FILE and DTC_OVERLAY_FILE may be a easy, convenient and acceptable way for a custom downstream application, but it's not a principle that should be used used in upstream samples.

A number of reasons, such as it's not scalable.
Users cannot opt-out of the forced `dts.overlay`.
Other flaws are seen here: #1522